docs/src/userguide/wrapping_CPlusPlus.rst

```diff
@@ -299,6 +299,14 @@ reference the type of a default template parameter for an incomplete template
 instantiation, it will write ``MyClass<T, U>::V``, so if the class provides
 a typedef for its template parameters it is preferable to use that name here.
 
+On assignments, Cython does not try to infer the template type of the
+newly created C++ object from a variable that it gets assigned to.
+The template type must be supplied  to both the constructor and the
```

Comment:
You don't have to have this redundancy, instead (and preferably) you can just write

    v = vector[int]()
    v_ptr = new vector[int]()

and type inference will do the right thing.